vis_gen.py

```diff
@@ -0,0 +1,565 @@
+        for idx, code_point in enumerate(code_points):
+            if idx % 50 == 0:
+                print("Now writing {}st code point.".format(idx + 1))
+            self.visualize_single(code_point, False, x=x, y=y)
+        print("Finished.")
+        print("Images stored in directory {}.".format(out_dir_abs))
+
+        self.__check_out_dir = True
+
+
+    def visualize_range(self, start, end, x=None, y=None):
+        """Render text images from start code point to end code point and write
+        to out_dir.
+        
+        Args:
+            start: Str, Unicode code point, starting code point to write
+            end: Str, Unicode code point, the last code point to write
+            x: Int, x coordinate of the position of text, in number of pixels
+            y: Int, y coordinate of the position of text, in number of pixels
+
+        Raises:
+            OSError: if specified directory cannot be created
+            TypeError: if start or end is not a single character code point
+        """
+        # Check if out_dir exists and create if not
+        out_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        if self.__check_out_dir:
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        # Compute ord for start and end code points
+        try:
+            ord_start = ord(start)
+            ord_end = ord(end)
+        except TypeError:
+            print("Expect start and end to be single character code point.")
+            raise
+
+        # Check start and end ord and exchange if necessary
+        if ord_start > ord_end:
+            ord_start, ord_end = ord_end, ord_start
+        
+        # Visualize all code points
+        total_cp = ord_end - ord_start + 1
+        print("Visualizing {} total code points from {} to {}."
+              .format(total_cp, start, end))
+        for cur_ord in range(ord_start, ord_end+1):
+            if (cur_ord - ord_start) % 50 == 0:
+                print("Now writing {}st code point."
+                      .format(str(cur_ord - ord_start + 1)))
+            code_point = chr(cur_ord)
+            self.visualize_single(code_point, False, x=x, y=y)
+        print("Finished.")
+        print("Images stored in directory {}.".format(out_dir_abs))
+
+        self.__check_out_dir = True
+
+
+    def visualize_single(self, code_point, check_out_dir=True, x=None, y=None):
+        """Write rendered text image for specific code point to output
+        directory, text is positioned at (x, y). If both x and y are set to None
+        (default), the text will be centered.
+
+        Args:
+            code_point: Str, single unicode code point
+            check_out_dir: Bool, set True if need to check output directory
+            x: Int, x coordinate of the position of text, in number of pixels.
+            y: Int, y coordinate of the position of text, in number of pixels.
+
+        Returns:
+            file_path: Str, path to file
```

Comment:
Since this is a public method, also document the file name format here as how you documented in __get_filename